esx_service/vmdk_ops.py

```diff
@@ -837,25 +838,31 @@ def executeRequest(vm_uuid, vm_name, config_path, cmd, full_vol_name, opts):
     except ValidationError as ex:
         return err(str(ex))
 
-    if datastore and not vmdk_utils.validate_datastore(datastore):
+    if not datastore:
+        datastore = default_datastore
+
+    # Its possible the datastore is None, which can happen if datastores on the
+    # host could not be identified or initialized with the dockvols path.
+    if not datastore:
```

Comment:
Why do you want to differentiate these two cases? And why the error message for these two cases printed out is different?